swarm/network/stream/stream.go

```diff
@@ -597,6 +597,16 @@ func (r *Registry) runProtocol(p *p2p.Peer, rw p2p.MsgReadWriter) error {
 
 // HandleMsg is the message handler that delegates incoming messages
 func (p *Peer) HandleMsg(ctx context.Context, msg interface{}) error {
+	select {
+	case <-p.streamer.quit:
+		log.Trace("message received after the streamer is closed", "peer", p.ID())
+		// return without an error since streamer is closed and
+		// no messages should be handled as other subcomponents like
+		// storage leveldb may be closed
+		return nil
```

Comment:
I tried to understand if it's OK to return `nil` or we should return an error. Well, I gave up. It's seems like I would have to understand half the world to decide on that.

Could you elaborate on what could go wrong if we return an error?

If I think about an HTTP API that should not return `200 OK` to the caller if the message was not handled. So in this case, I think the caller also should know we did not handle the message.

Reply:
From the handling node (the receiver of messages), if we return an error, we will log it with Error severity, which is not something that would be an actual error. From the client point of view, we will just not respond with no RPC message response. This is an RPC interface, so no HTTP codes or responses should be involved.